### DISCUSSION TITLE
VRP: enable different distance limit for different types of vehicle

### DISCUSSION THREAD

**Discussion by Westlife1002** [google/or-tools#2731] [Routing (and legacy CP) questions] [answered]
How to set differnt max driving distance for different types of vehicle?
e.g. If I have large, medium, and small vehicles, I want to set max of 180km for small vehicle, 210km for medium vehicle, and 250 for large vehicle.

previously, the max driving distance limit is set all the same via below function:

```python
def add_distance_dimension(routing, transit_callback_index, data, manager):
	"""Add Global Span constraint"""
	distance = "Distance"
	maximum_distance = 280
	routing.AddDimension(
		transit_callback_index,
		0, # null slack
		maximum_distance, # maximum distance per vehicle
		True, # start cumul to zero
		distance)
```

**Comment by Mizux** [comment, marked as answer]
Use the overload function `AddDimensionWithVehicleCapacity()`

ref: https://github.com/google/or-tools/blob/b37d9c786b69128f3505f15beca09e89bf078a89/ortools/constraint_solver/routing.h#L449-L452

**Reply by Westlife1002** [reply]
yes, it worked well. thank you
